### PR TITLE
Justus Exercises_07 

### PR DESCRIPTION
**Pull request by Saiki77**
https://github.com/user-attachments/assets/acf047e9-d4d0-468d-b522-d6540d73d1bb

### CODE REVIEW

**Comment by Saiki77** [comment]
My final score is: 17159

File: Exercises/Justus Exercises/Justus Exercises_07.py [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+                1: "#FF7F00",        # Second row: Orange
+                2: "#FFFF00",        # Third row: Yellow
+                3: "#00CC00",        # Fourth row: Green
+                4: "#0000FF",        # Bottom row: Blue
+                5: "#4B0082",        # Additional row: Indigo
+                6: "#9400D3"         # Additional row: Violet
+            },
+            self.PADDLE: "green",    # Paddle remains green
+            self.WALL: "white"       # Walls remain white
+        }
```
Comment (FabianPerl93) [inline Exercises/Justus Exercises/Justus Exercises_07.py:40]: I like your colourfull design.

File: Exercises/Justus Exercises/Justus Exercises_07.py [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+
+    @staticmethod
+    def parse_instruction(instruction: int) -> Tuple[int, List[int]]:
+        """Parse an instruction into opcode and parameter modes."""
+        opcode = instruction % 100
+        modes = [(instruction // 10 ** i) % 10 for i in range(2, 5)]
+        return opcode, modes
+
+
+    def run_game(self):
```
Comment (FabianPerl93) [inline Exercises/Justus Exercises/Justus Exercises_07.py:142]: Implementing both the option of automatic and manual mode is a nice idea.

File: Exercises/Justus Exercises/Justus Exercises_07.py [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+            with open(file_path, 'r') as file:
+                content = file.read().strip()
+                # Handle both comma-separated and newline-separated inputs if the input changes again 
+                content = content.replace('\n', ',')
+                return [int(x) for x in content.split(',')]
+        except FileNotFoundError:
+            raise FileNotFoundError(f"Memory file not found: {file_path}")
+        except ValueError:
+            raise ValueError("Invalid memory file format: must contain comma-separated integers")
+
```
Comment (FabianPerl93) [inline Exercises/Justus Exercises/Justus Exercises_07.py:55]: nice custom error messages.

**Review by FabianPerl93 — APPROVED** [review]
The live score counter is a cool feature

**Pull request merged**